text/src/autogluon/text/automm/utils.py

```diff
@@ -503,6 +508,72 @@ def create_model(
     else:
         raise ValueError(f"No available models for {names}")
 
+def save_pretrained_model(
+    model: ModuleList,
+    config: DictConfig,
+    path: str
+) -> DictConfig:
+    '''
+    Saving the pretrained weights for offline deployment. Used by turing "standalone=True" in predictor. 
+
+    Parameters
+    ----------
+    model
+        A modulelist containing all models.
+    config
+        A DictConfig object. The model config should be accessible by "config.model".
+    path
+        The saving path to the pretained weights i.e. config.json for "clip" and "hf_text", 'pkl' for "timm_image"
+    '''
+    for idx, model_name in enumerate(config.model.names):
+        if model_name == "clip" or "hf_text" in model_name:
+            model[idx].model.save_pretrained(os.path.join(path,model_name))
+            model_config = getattr(config.model, model_name)
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+        if model_name == "timm_image":
+            model_config = getattr(config.model, model_name)
+            if not os.path.exists(os.path.join(path,model_name)):
+                os.makedirs(os.path.join(path,model_name))
+            torch.save(
+                model[idx].model.state_dict(),
+                os.path.join(path,model_name,model_config.checkpoint_name+'.pkl')
+            )
+            model_config.checkpoint_name = os.path.join('local://',model_name)
+    return config
+
+
+def load_pretrained_configs(
+    config: DictConfig,
+    path: str
+) -> DictConfig:  
+    '''
+    Load the pretrained weights for offline deployment. Used by turing "standalone=True" in predictor. 
+
+    Parameters
+    ----------
+    config
+        A DictConfig object. The model config should be accessible by "config.model".
+    path
+        The saving path to the pretained weights i.e. config.json for "clip" and "hf_text", 'pkl' for "timm_image"
+    '''
+    setattr(config,'pretrained',True)
+    for model_name in config.model.names:
+        if model_name == "clip" or "hf_text" in model_name:
+            model_config = getattr(config.model,model_name)
+            if model_config.checkpoint_name.startswith('local://'):
+                model_config.checkpoint_name = os.path.join(path,model_config.checkpoint_name[len('local://'):])
+                assert os.path.exists(os.path.join(model_config.checkpoint_name,'config.json'))
+                assert os.path.exists(os.path.join(model_config.checkpoint_name,'pytorch_model.bin'))
```

Comment:
Where do we use 'pytorch_model.bin' and 'config.json'?

Reply:
I think we are safe to remove the check here.

Reply:
Both 'pytorch_model.bin' and 'config.json' are used by calling `create_model` in AutoMMPredictor.load with a checkpoint referring to the local dir where they are stored. To be more specific, in `create_model` of AutoMMPredictor.load  [https://github.com/yiqings/autogluon/blob/8985d83bf238432ebba5062655a7a4b45bea44d3/text/src/autogluon/text/automm/predictor.py#L971](url), config.model.{MODEL NAME}.checkpoint_name is a local dir pointing to 'pytorch_model.bin' and 'config.json'.

Reply:
The assertion guarantees the existence of 'pytorch_model.bin' and 'config.json'. The missing of one of them will raise an error in loading the models in `create_model`.